src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class UnassignedFieldsWalker : DataFlowPass
+    {
+        private readonly DiagnosticBag _diagnostics;
+
+        private UnassignedFieldsWalker(CSharpCompilation compilation, MethodSymbol method, BoundNode node, DiagnosticBag diagnostics)
+            : base(compilation, method, node, trackClasses: true)
+        {
+            _diagnostics = diagnostics;
+        }
+
+        internal static void Analyze(CSharpCompilation compilation, MethodSymbol method, BoundNode node, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(method.MethodKind == MethodKind.Constructor);
+
+            var flags = ((CSharpParseOptions)node.SyntaxTree.Options).GetNullableReferenceFlags();
```

Comment:
If we do such checks in multiple places in the code (prior to this PR), consider introducing a helper method.